src/NHibernate/Util/ReflectionCache.cs

```diff
@@ -67,6 +67,11 @@ internal static class QueryableMethods
 		{
 			internal static readonly MethodInfo SelectDefinition =
 				ReflectHelper.GetMethodDefinition(() => Queryable.Select(null, default(Expression<Func<object, object>>)));
+			internal static readonly MethodInfo SelectManyDefinition =
+				ReflectHelper.GetMethodDefinition(() => Queryable.SelectMany<object, object, object>(null, (Expression<Func<object, IEnumerable<object>>>) null, null));
+
+			internal static readonly MethodInfo GroupJoinDefinition =
+				ReflectHelper.GetMethodDefinition(() => Queryable.GroupJoin<object, object, object, object>(null, null, null, null, null));
```

Comment:
These would have to be updated if #2352 gets merged before, or conversely, #2352 would have to be updated on these if #2328 gets merged before.

Reply:
Rebased and updated.